src/formats/notion-api/api-helpers.ts

```diff
@@ -388,6 +387,29 @@ export async function extractFrontMatter(
 	return frontMatter;
 }
 
+/**
+ * Extract notion timestamps from file frontmatter
+ * Used for delta sync to determine if file needs updating
+ * @param fileContent - The raw file content including frontmatter
+ * @returns Object with created and updated timestamps, or null if not found
+ */
+export function extractNotionTimestamps(fileContent: string): {
```

Comment:
Instead of pulling this out of the file content using string matching, please use the metadata cache to read it from the parsed frontmatter.

Change the arguments to accept `app: App`, then

```ts
app.metadataCache.getFileCache(file).frontmatter...
```